src/main/js/portal/main/components/preview/PreviewTimeSerie.tsx

```diff
@@ -167,14 +158,20 @@ export default class PreviewTimeSerie extends Component<OurProps, OurState> {
 
 		const showChartTypeControl = !previewExcludeChartType.datasets.includes(preview.item.dataset!);
 
-		const yParam = yAxis ? `&y=${yAxis}` : '';
-		const y2Param = y2Axis ? `&y2=${y2Axis}` : '';
-		const legendLabelsParams = legendLabels ? `&legendLabels=${legendLabels}` : '';
-		const legendLabelsY2Params = y2Axis && legendLabels ? `&legendLabelsY2=${legendLabels}` : '';
-		const iframeUrl = `${window.document.location.protocol}//${window.document.location.host}${iFrameBaseUrl}?objId=${objIds}&x=${xAxis}${yParam}${y2Param}&type=${type}&linking=${linking}${legendLabelsParams}${legendLabelsY2Params}`;
-
-		if (!preview)
-			return null;
+		const iframeParams = [`?objId=${objIds}`,
+			`&x=${xAxis}`,
+			yAxis ? `&y=${yAxis}` : '',
+			y2Axis ? `&y2=${y2Axis}` : '',
+			`&linking=${linking}`,
+			legendLabels ? `&legendLabels=${legendLabels}` : '',
+			y2Axis && legendLabels ? `&legendLabelsY2=${legendLabels}` : '',
+			type ? `&type=${type}` : '',
+		];
+
+		const iframeUrl = `${window.document.location.protocol}//`
+			+ `${window.document.location.host}`
+			+ `${iFrameBaseUrl}`
+			+ `${iframeParams.join('')}`;
```

Comment:
Sorry to comment on this again, but I read your comments and had some more thoughts on this. Since there's been some efforts improving this code, I think we can go even further. I believe the following suggestion is equivalent, but I didn't test it fully.
`iframeUrl' is now a 'URL', so if it's important that it's a string, the two usages further down needs a `toString()` added.

```suggestion
		let iframeUrl = new URL(iFrameBaseUrl, window.document.location.origin)

		iframeUrl.searchParams = new URLSearchParams([
      ['objId', objIds], 
      ['type', type], 
      ['linking', linking],
      ['x', xAxis]
    ])

    if (yAxis) {
      iframeUrl.searchParams.append('y', yAxis)
    }

    if (y2Axis) {
      iframeUrl.searchParams.append('y2', y2Axis)
    }

    if (legendLabels) {
      iframeUrl.searchParams.append('legendLabels', legendLabels)
    }

    if (y2Axis && legendLabels) {
      iframeUrl.searchParams.append('legendLabelsY2', legendLabelsY2)
    }
```

Reply:
Yeah, wow, I didn't know the `URLSearchParams` had such overhead. I thought it was basically a `Map`.
Anyway, my last contribution is this, which in my test is the fastest :)
My main input, though, was supposed to be that I think it's clearer to explicitly set the conditional things in if-blocks (as in my first suggestion and this one), rather than setting fields to null or empty and then filtering them later. 

```js
    let params = {
      objId: objIds,
      x: xAxis,
      linking,
      type
    };

    if (yAxis) {
      params.y = yAxis
    }

    if (y2Axis) {
      params.y2 = y2Axis
    }

    if (legendLabels) {
      params.legendLabels = legendLabels
    }

    if (y2Axis && legendLabels) {
      params.legendLabelsY2 = legendLabelsY2
    }

    const iframeParamsStr = Object.entries(params)
      .map((param) => param.join('='))
      .join("&");
```

Reply:
Ah, I see. It's rather verbose but definitely more readable and clear. The type definition ends up a bit odd (`let params: PreviewSettings & { objId: string }`), but it is definitely a nice improvement, and I've added it in.